homeassistant/components/onewire/binary_sensor.py

```diff
@@ -98,15 +112,15 @@ def get_entities(onewirehub: OneWireHub):
 
         if family not in DEVICE_BINARY_SENSORS:
             continue
-        device_info = {
+        device_info: DeviceInfo = {
             "identifiers": {(DOMAIN, device_id)},
             "manufacturer": "Maxim Integrated",
             "model": device_type,
             "name": device_id,
         }
         for entity_specs in DEVICE_BINARY_SENSORS[family]:
             entity_path = os.path.join(
-                os.path.split(device["path"])[0], entity_specs["path"]
+                os.path.split(device["path"])[0], str(entity_specs["path"])
```

Comment:
Shouldn't be needed with `TypedDict`.
```suggestion
                os.path.split(device["path"])[0], entity_specs["path"]
```